src/core/raster/qgssinglebandpseudocolorrenderer.cpp

```diff
@@ -84,6 +88,193 @@ void QgsSingleBandPseudoColorRenderer::setShader( QgsRasterShader* shader )
   mShader = shader;
 }
 
+void QgsSingleBandPseudoColorRenderer::createShader( QgsColorRamp* colorRamp, QgsColorRampShader::ColorRamp_TYPE colorRampType, bool clip )
```

Comment:
Good point... I think it should go to QgsColorRampShader.

Thinking about it a bit more, it probably also makes sense to actually store also other bits regarding classification directly in QgsColorRampShader. We already store there the source color ramp which is not used for rendering, so why not also the classification mode...

Probably we do not even need to explicitly store number of classes in the renderer - we can just find out the count from the number of color ramp entries, right? We also may not need to store min/max values in the renderer as they are also available from the color ramp shader instance :-)

(This is already outside of the scope of this PR, but it may be useful to get rid of QgsRasterShaderFunction - the base class of QgsColorRampShader. The color ramp shader is the only sub-class and the base class is only ever used in pseudo color renderer anyway. It could be a nice cleanup for 3.0 since all the code only expects QgsColorRamp objects where QgsRasterShaderFunction can be.)